core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SubmissionApiServiceSpec.scala

```diff
@@ -986,4 +987,130 @@ class SubmissionApiServiceSpec extends ApiServiceSpec with TableDrivenPropertyCh
     }
   }
 
+  private val userComment1000character = RandomStringUtils.random(1000)
+  private val validUserCommentCases = Table(
+    ("description", "userCommentInput", "userCommentResult"),
+    ("allow submission with userComment unset", None, JsNull),
+    ("allow submission with userComment", Option("This submission outputs hello world. Cost: $0.5"), JsString("This submission outputs hello world. Cost: $0.5")),
+    ("allow submission with userComment that has special characters", Option("This comment has special characters: `~!@#$%^&*()_+-=[]\\{}|;':\",./<>?"), JsString("""This comment has special characters: `~!@#$%^&*()_+-=[]\{}|;':",./<>?""")),
+    ("allow submission with userComment that has unescaped special characters", Option("""This comment has special characters: `~!@#$%^&*()_+-=[]\{}|;':",./<>?"""), JsString("""This comment has special characters: `~!@#$%^&*()_+-=[]\{}|;':",./<>?""")),
```

Comment:
Can we test some non-ASCII special chars like (just mashing my keyboard here) •ªº§¶œ∑´©˙∆˚˜∫µ˜≤